### ISSUE TITLE
Stan errors contains terminal color codes

### ISSUE THREAD

**Issue by andys8** [haskell/haskell-language-server#3089]
## Is your enhancement request related to a problem? Please describe.

Severity and Category use terminal color codes that aren't recognized in Vim/CoC. I assume that's also the case in other editors.

![image](https://user-images.githubusercontent.com/13085980/184101907-8bc89dbc-dab5-4be3-aabb-59cd6cda636b.png)

## Describe the solution you'd like

* Either allowing `stan` itself to configure/detect non interactive mode or skip coloring
* Or filter out color codes in the stan hls plugin

## Additional context

Currently using `a89ba412293327ec838c00d899ea9c6013c46aa3`.

```shell
haskell-language-server version: 1.7.0.0 (GHC: 9.2.4) (PATH: /home/andreas/.ghcup/bin/haskell-language-server-wrapper-1.7.0.1)
Tool versions found on the $PATH
cabal:          3.8.1.0
stack:          2.7.5
ghc:            9.2.4
```

In case the local stan installation is relevant:

```
Stan v0.0.0.0
 ➤ Git revision: 4fcf2e15d067f652c753272202fb33ef4200ccad
 ➤ Commit date:  Thu Jun 25 18:40:45 2020 +0100
```

Pinging @uhbif19

**Comment by andys8** [comment]
It's easier than I initially thought. Pretty functions are adding terminal colors and are directly used. Replacing them with show instances solves the issue.

* https://github.com/kowainik/stan/blob/dcbc8682a964ae6f642ae37adbd46f6545d39aa6/src/Stan/Category.hs#L36-L38
* https://github.com/kowainik/stan/blob/dcbc8682a964ae6f642ae37adbd46f6545d39aa6/src/Stan/Severity.hs#L74-L76
* 
Opened #3090

**Comment by andys8** [comment]
Solved by #3090
